specs/RestartRequested.md

```diff
@@ -70,14 +124,26 @@ interface ICoreWebView2RestartRequestedEventHandler : IUnknown {
 [uuid(ef7632ec-d86e-46dd-9d59-e6ffb5c87878), object, pointer_default(unique)]
 interface ICoreWebView2Environment10 : IUnknown {
   /// Add an event handler for the `RestartRequested` event.
-  /// `RestartRequested` event is raised when there is an urgent need to 
-  /// restart the WebView2 process in order to enable or disable 
-  /// features that are causing WebView2 reliability or performance to drop critically.
-  /// `RestartRequested` gives you the awareness of these necessary WebView2 restarts,
-  /// allowing you to resolve issues faster than waiting for your end users to restart the app.
-  /// Depending on your app you may want to prompt your end users in some way to give
-  /// them a chance to save their state before you restart the WebView2.
+  /// `RestartRequested` event is raised when there is a need to restart WebView2 process
+  /// in order to apply certain beneifical updates.
+  /// Those updates can be the following cases:
+  /// 1. New runtime version available.
+  /// 2. Older runtime version available (downgradge due to severe bug).
+  /// 3. Features being enabled/disabled that causing WebView2 reliability or performance 
+  ///    to drop dramatically. 
+  /// 
+  /// `RestartRequested` is raised base on priority level. `RestartPriorityLevel` is used
+  /// to determine the urgences on rasing such event.
+  /// `COREWEBVIEW2_RESTART_PRIORITY_LEVEL_BEST_EFFORT`:
+  ///   - Event rasied only on new runtime version available.
+  ///   - Developer can notify user to restart on normal cadence.
+  /// `COREWEBVIEW2_RESTART_PRIORITY_LEVEL_CRITICAL`: 
+  ///   - Event raised for downgraded runtime version and feature enabled/disabled.
```

Comment:
For this and line 138 - we don't want to document that these enum values correspond directly to best effort = upgrade, and critical = downgrade or configuration change. We want to be able to change that in the future